### PR TITLE
Fix empty links

### LINKED ISSUE #3661: Some anchor elements have empty href attributes
## Expected Behavior

All anchor elements with `href` attributes have a value in that attribute.

## Actual Behavior

Some anchor elements have `href=""`. htmlproofer is catching those now in automated tests.

## Steps to reproduce the behavior

- build the site

## This issue is done when

- [ ] htmlproofer tests show 0 errors for empty `href` attributes

### PR DESCRIPTION
**Pull request by mgwalker**
# Pull request summary

For links with an empty `href` attribute and no `id`, removes the link or changes `name` to `id`. Closes #3661.

### CODE REVIEW

File: _posts/2017-09-18-a-day-in-the-life-of-an-18f-product-owner.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ We interviewed our product owner, Aaron Burk, for our Forest Service [engagement
...
+**Aaron Burk:** As a product owner, my role is to ensure that our partners at GSA/TTS/18F have the information necessary to put together the appropriate solicitation documents as we start to engage contractors selected from the Agile Blanket Purchase Agreement. As we work through that process, the product owner needs to be able to weigh in and help evaluate the results of the bids we receive in relation to the work that we need done, as we have for our first two buys, the logic layer [Application Program Interface](https://github.com/18F/fs-middlelayer-api) and the [intake module](https://github.com/18F/fs-intake-module). Once we select a vendor, my role is a continued relationship with the vendor, alongside TTS, to keep the communications open to get the product that the Forest Service requires to best serve the public’s needs. I have wonderful resources from the TTS team to help with the technical aspects of the project. It’s key to leverage the resources we have but also strongly communicate the values of the agency and the users of the system.
```
Comment (mgwalker) [inline _posts/2017-09-18-a-day-in-the-life-of-an-18f-product-owner.md:33]: Removed a link to the Agile Blanket Purchase Agreement. Apparently we've removed all references to it from our site, except for blog posts. Maybe we should link to [this blog post](https://18f.gsa.gov/2018/07/26/what-we-learned-from-building-a-pool-of-agile-vendors/) where we kind of do a retro on the Agile BPA after we sunset it?

File: _principles/index.md [hunk 2]
```diff
@@ -33,15 +33,15 @@ In this document, we’ll explain each idea and how it will influence our work t
 
 # Methodologies
 
-## <a name="letting-users-guide-our-work"></a>Letting users guide our work
```
Comment (mgwalker) [inline _principles/index.md:36]: Jekyll adds landmark IDs to headings automatically, so we don't need to do it explicitly with these anchors. Plus the `name` attribute is deprecated now.

File: _authors/mel-choyce.md [hunk 3]
```diff
@@ -3,5 +3,6 @@ name: mel-choyce
 first_name: Mel
 last_name: Choyce
 full_name: Mel Choyce
-published: false
+published: true
```
Comment (mgwalker) [inline _authors/mel-choyce.md:6]: Everytime I build the site, it updates this file, so I figure it's probably supposed to be updated. I'm just gonna commit it.

**Review by colinmurphy01 — APPROVED** [review]
👍 

Checked the PR and logs, let's get this money.

**Pull request merged**